### PR TITLE
Support annotation use site targets

### PR DESCRIPTION
**Pull request by Egorand**
Implements #150.

#153 provides cleaner emission logic, once it's merged I'll update the PR.

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import java.io.IOException
 import java.io.StringWriter
 import java.lang.reflect.Array
 import java.util.Arrays
+import java.util.Locale
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt]: I don't want to add any new dependencies on the JVM. An easy way around this is to add the raw output string to each enum value.

Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt]: Makes sense. `KModifier` uses the same approach to printing, worth opening an issue to fix that as well?

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt]: Sure. Or just do a PR if you have time.

I think long-term we'll have a core KotlinPoet and then platform-specific helpers for things like java.lang.Class, TypeMirror, TypeElement, Filer, etc. in a kotlinpoet-jvm artifact.

Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt]: Pushed the fix

**Comment by JakeWharton** [comment]
I'll rebase on you

**Comment by Egorand** [comment]
Great, thanks!

**Pull request merged**